<?xml version="1.0"?>
<launch>
    <group ns="agv">
        <node name="node_lidar" pkg="rplidar_ros" type="rplidarNode" output="screen">
            <param name="serial_port"         type="string" value="/dev/ttyUSB_LIDAR"/>
            <param name="serial_baudrate"     type="int"    value="256000"/>
            <param name="frame_id"            type="string" value="base_scan"/>
            <param name="inverted"            type="bool"   value="false"/>
            <param name="angle_compensate"    type="bool"   value="true"/>
        </node>

        <node name="node_imu" pkg="rosserial_python" type="serial_node.py" output="screen">
            <param name="port" value="/dev/ttyUSB_ARDUINO"/>
            <param name="baud" value="115200"/>
            <remap from="/imu/rpy" to="/agv/imu/rpy"/>
        </node>

        <node name="node_mcu" pkg="rosserial_python" type="serial_node.py" output="screen">
            <param name="port" value="/dev/ttyUSB_MCU"/>
            <param name="baud" value="115200"/>
        </node>

        <node name="node_uwb" pkg="agv" type="dwm1001_main.py" output="screen">
            <param name="serial_port_name" value="/dev/ttyUSB_UWB"/>
            <param name="serial_baud_rate" value="115200"/>
            <remap from="/dwm1001/tag" to="/agv/uwb/tag"/>
            <remap from="/dwm1001/anchor0" to="/agv/uwb/anchor0"/>
            <remap from="/dwm1001/anchor1" to="/agv/uwb/anchor1"/>
            <remap from="/dwm1001/anchor2" to="/agv/uwb/anchor2"/>
            <remap from="/dwm1001/anchor3" to="/agv/uwb/anchor3"/>
        </node>

        <!-- <node name="node_uwb_display" pkg="agv" type="dwm1001_display.py"  output="screen"/> -->
    </group>
</launch>